// skyflow-node/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Skyflow = exports.LogLevel = exports.setLogLevel = exports.generateBearerTokenFromCreds = exports.GenerateToken = exports.generateBearerToken = void 0;
const Token_1 = require("./service-account/util/Token");
Object.defineProperty(exports, "GenerateToken", { enumerable: true, get: function () { return Token_1.GenerateToken; } });
Object.defineProperty(exports, "generateBearerToken", { enumerable: true, get: function () { return Token_1.generateBearerToken; } });
Object.defineProperty(exports, "generateBearerTokenFromCreds", { enumerable: true, get: function () { return Token_1.generateBearerTokenFromCreds; } });
const Skyflow_1 = __importDefault(require("./vault-api/Skyflow"));
exports.Skyflow = Skyflow_1.default;
const Logging_1 = require("./vault-api/Logging");
Object.defineProperty(exports, "setLogLevel", { enumerable: true, get: function () { return Logging_1.setLogLevel; } });
const common_1 = require("./vault-api/utils/common");
Object.defineProperty(exports, "LogLevel", { enumerable: true, get: function () { return common_1.LogLevel; } });
